libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/roadname/view/MapboxRoadNameView.kt

```diff
@@ -43,16 +43,24 @@ class MapboxRoadNameView : AppCompatTextView {
         defStyleAttr: Int
     ) : super(context, attrs, defStyleAttr)
 
+    /**
+     * The method is invoked automatically when the view is detached from the window and clears any
+     * relevant in memory cache.
+     */
+    override fun onDetachedFromWindow() {
+        shields.clear()
+        roadComponents.clear()
+        super.onDetachedFromWindow()
+    }
```

Comment:
What about the case where an app runs for a very long time? `onDetachedFromWindow` won't be called, so we will continue to accumulate shields. Can this lead to a noticeable memory consumption?
Also `onDetachedFromWindow` is usually called in `Activity.onDestroy` or `Fragment.onDestroyView`, when the view is going to be destroyed anyway, so clearing shields won't help that much. 
Maybe we should just use `LruCache`, which will automatically remove the shields that are no longed used.

Reply:
Thanks for bringing this up :+1: 

`shields` is a `set`, and there are not that many shields encountered during a nav session, so I wouldn't consider this a risk but definitely a worthy optimization in general.

Thing is, the shield references that are being stored here are going to be stored in the `RoadShieldContentManagerContainer` anyway, which is an `object` so they would be stored across the processes lifetime. Again, we didn't consider this a risk due to the relatively small size and low frequency of shields but we should definitely think about a wider, more generic initiative of setting up `LruCache` for resources that SDK pre-downloads and stores in-memory for the duration of the session/process. Part of that is going to be tackled by the Common Native Cache initiative but we might want to migrate the process-long cache to `LruCache` just for safety against potential edge cases where the app doesn't restart for weeks or months. Tracking in https://github.com/mapbox/mapbox-navigation-android/issues/5418.

Reply:
I think we can remove these `onDetachedFromWindow` overrides then. Because it will help only in cases where the view is kept after being manually removed from its parent, so that it can be re-added later. I doubt anyone will do this for these particular views and even if they do, the help is not significant, because the shields themselves will be retained anyway.

Reply:
Agreed :+1: